### PR TITLE
[Kotlin Migration] DeckAdapter

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    private inner class DeckFilter : Filter() {
+        private val mFilteredDecks = ArrayList<AbstractDeckTreeNode<*>>()
+        private val allDecks: List<T>
+            get() = mDeckList
+
+        override fun performFiltering(constraint: CharSequence): FilterResults {
+            mFilteredDecks.clear()
+            mFilteredDecks.ensureCapacity(mCol!!.decks.count())
+            if (TextUtils.isEmpty(constraint)) {
+                mFilteredDecks.addAll(allDecks as ArrayList<AbstractDeckTreeNode<*>>)
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.kt:316]: unchecked conversion error, what should I do?

File: AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+            } else {
+                val filterPattern = constraint.toString().lowercase(Locale.getDefault()).trim { it <= ' ' }
+                val filteredDecks = filterDecks(filterPattern, allDecks)
+                mFilteredDecks.addAll(filteredDecks as ArrayList<AbstractDeckTreeNode<*>>)
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/widgets/DeckAdapter.kt:320]: same here,

in previous java file conversion to "ArrayList<AbstractDeckTreeNode<*>" was not required

**Comment by david-allison-1** [comment]
This was a horrible one - save it for later

**Comment by david-allison** [comment]
@Akshay0701 Informational only: https://github.com/ankidroid/Anki-Android/pull/10881

The trick was fixing it in the header: `DeckAdapter<T : AbstractDeckTreeNode<T>>`

**Comment by Akshay0701** [comment]
Oh Thanks!

**Pull request closed without merging**